src/test/java/io/r2dbc/postgresql/message/backend/BackendMessageDecoderTest.java

```diff
@@ -17,71 +17,61 @@
 package io.r2dbc.postgresql.message.backend;
 
 import io.netty.buffer.ByteBuf;
+import io.netty.buffer.CompositeByteBuf;
 import org.junit.jupiter.api.Test;
-import reactor.core.publisher.Flux;
-import reactor.core.publisher.Mono;
-import reactor.test.StepVerifier;
-import reactor.test.StepVerifier.FirstStep;
 
-import java.time.Duration;
 import java.util.Collections;
 import java.util.function.Function;
-import java.util.stream.Stream;
 
 import static io.netty.util.CharsetUtil.UTF_8;
 import static io.r2dbc.postgresql.message.Format.FORMAT_BINARY;
 import static io.r2dbc.postgresql.message.backend.Field.FieldType.CODE;
 import static io.r2dbc.postgresql.message.backend.ReadyForQuery.TransactionStatus.IDLE;
 import static io.r2dbc.postgresql.util.TestByteBufAllocator.TEST;
-import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.fail;
 
 final class BackendMessageDecoderTest {
 
     @Test
     void authenticationCleartextPassword() {
-        decode('R', buffer -> buffer.writeInt(3))
-            .expectNext(AuthenticationCleartextPassword.INSTANCE)
-            .verifyComplete();
+        BackendMessage message = decode('R', buffer -> buffer.writeInt(3));
+        assertThat(message).isEqualTo(AuthenticationCleartextPassword.INSTANCE);
     }
 
     @Test
     void authenticationGSS() {
-        decode('R', buffer -> buffer.writeInt(7))
-            .expectNext(AuthenticationGSS.INSTANCE)
-            .verifyComplete();
+        BackendMessage message = decode('R', buffer -> buffer.writeInt(7));
+        assertThat(message).isEqualTo(AuthenticationGSS.INSTANCE);
     }
 
     @Test
     void authenticationGSSContinue() {
```

Comment:
Every `BackendMessage` with `java.nio.ByteBuffer` fields can't be processed after `BackendMessageDecoder` called `release` on envelope buf since internal it's calling `ByteBuffer.duplicate` but clears the memory on calling `ByteBuf.release` (`UnpooledUnsafeDirectByteBuf.nioBuffer` and `UnpooledUnsafeDirectByteBuf.deallocate`).